servers/shares-api/src/apollo/context.ts

```diff
@@ -16,8 +22,27 @@ export async function getContext({
   });
 }
 
-export interface IContext {}
+export interface IContext {
+  PocketShareModel: PocketShareModel;
+}
 
 export class ContextManager implements IContext {
-  constructor(options: { request: Request }) {}
+  PocketShareModel: PocketShareModel;
+  constructor(options: { request: Request }) {
+    const userId = options.request.headers.userid;
+    const client = dynamoClient();
+    // We have an authenticated user?
+    if (
+      userId &&
+      typeof userId === 'string' &&
+      userId.length &&
+      userId !== 'anonymous'
```

Comment:
lets ticket making this logic and logic from the context headers into a package.

Reply:
https://mozilla-hub.atlassian.net/browse/POCKET-9973